### PR TITLE
New Away Mission map

### PR DESCRIPTION
**Pull request by OceanFish1**
## About The Pull Request

Hey hey people, Fish here. Today I'll be pulling a new away mission I was making for months. It's called "The Factory". So, this is the kind of mission where you supposed to get tons of gamer loot, but except one thing. You don't get it for free this time. You had too much for previous missions (except underground outpost which is usually used for plasmaman erotic role play and burning everyone alive). You're gonna struggle to get your gamer loot, but be sure it worth it. The map size is 225x210 and only one z-level. Why only one? Because we fuck multi-z level maps, we don't do that here. Instead we do tons teleportation to make the player feel they're actually going up or down.
Now the difficulty of the mission. This contains good amount of mobs, where most of them can shoot. Some of them can shoot your limb or your brains off and two of them may husk you instantly. The good thing is that they're all lootable, which means when the die they don't magically turn into human gibs. They leave bodies with all the things they had when they were alive.
Now about the plot. 
Who the fuck needs a plot for an away mission? You just go in and do whatever. I may already broke your aesthetic feelins, so here's your plot: A group of people on Earth decided to make a company that specializes on finding ways to produce infinity food. They found some god's forgotten village and built a big building. They decided that villagers should not know what's happening inside and made the outer walls look like it's made out of wood. Then they built an underground facility. Then they went too far and built a gateway, the same like on the station. They made a special squad for gateway expeditions that is tasked to go in, grab whatever they see, get bonus points if it's behaving way too strange for people and bring it to factory for """""research""""". They were stealing shit from wizards, nanotrasen, other people. Your mission goal is to go """""""talk""""""" with them and ask them to stop stealing. All you need to do for this is to travel in the past, when they didn't abandon the place. Yes, you gotta time travel and """"""ask"""""" their boss to turn off the gates. Sounds easy.

Since it's the first PR I ever made and the first attempt to code I won't be surprized if I already commited several warcrimes while was writing this. But at least I found myself as a (good) mapper. I even got to do some sprites. Yes, that pile of wood is actually a shotgun. Most of the sounds are already existing sounds, but edited. Some of them are mine.
 If things went wrong please tell me about them. Especially about the balance.

factory

## Why It's Good For The Game

The previous missions are outdated as fuck and thats why they're being reworked. But while it's not done yet, I wanna pull my own mission because I don't see a reason why not.

## Changelog
:cl:
add: New away mission
add: New mobs and items for new away mission
soundadd: added few sounds for away mission mobs and ambience
imageadd: added icons for a lever-action shotgun, tactical sunglasses and some mobs
code: made gatling laser emaggable
config: added The Factory to away missions
/:cl:

### CODE REVIEW

**Comment by Fronsis** [comment]
Looks very nice, you should check with Cae to have it added once his rework is finished

**Comment by OceanFish1** [comment]
> Looks very nice, you should check with Cae to have it added once his rework is finished

Yeah, we've been talking about that.

**Comment by zeskorion** [comment]
ha ah ha ha hah aha hahahahahhahahahhhhhhhaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaa

**Comment by PowerfulBacon** [comment]
When it comes to gamer loot, there is no such thing as hard to get gamer loot. There are always over 100 ways to cheese the threat in a way you haven't thought of.

**Comment by OceanFish1** [comment]
> When it comes to gamer loot, there is no such thing as hard to get gamer loot. There are always over 100 ways to cheese the threat in a way you haven't thought of.

Of course I know that, but first I need to see what in ways the threat will be avoided. There is also a thing that sometimes the threat IS the loot.

**Review by PowerfulBacon — DISMISSED** [review]
Incorrect map format. Must use TGM

**Comment by knacker48** [comment]
The key isn't to make it uncheeseable, its to make it hard enough that you need to put in a lot of effort to cheese. If you have a geneticist with tele and xray attacking your mobs with a crowbar two rooms away, make your stronger guys teleport to your position or beeline to you, smashing walls along the way.

**Comment by PowerfulBacon** [comment]
Don't include a_gifts

**Comment by OceanFish1** [comment]
> Don't include a_gifts

also why no gifts? They may give something game-breaking or what?

**Comment by PowerfulBacon** [comment]
> also why no gifts? They may give something game-breaking or what?

They can give any item in the game, including inaccessible things (Hence why only available on 1 day of the year for the memes).

**Comment by OceanFish1** [comment]
> > also why no gifts? They may give something game-breaking or what?
> 
> They can give any item in the game, including inaccessible things (Hence why only available on 1 day of the year for the memes).

understandable

**Comment by ItsTisme** [comment]
iam a sucker for factory aesthatic, would be nice if a away mision would be actuel production facility

**Comment by OceanFish1** [comment]
> iam a sucker for factory aesthatic, would be nice if a away mision would be actuel production facility

it has conveyor belts that are going circle isn't that enough?????///

**Comment by Kryyto** [comment]
The "time travel" mechanic work well for me, the mob have some nice behavior (speaking and screaming new dialogues line for example). And they are indeed op, found it cool

File: code/modules/projectiles/guns/ballistic/laser_gatling.dm
Comment (utilbird) [inline code/modules/projectiles/guns/ballistic/laser_gatling.dm]:
```suggestion
			if(cooldown < world.time)
```

Comment (OceanFish1) [reply code/modules/projectiles/guns/ballistic/laser_gatling.dm]: i don't like that too so i killed space

**Comment by TheFakeElon** [comment]
looks good

File: code/modules/projectiles/guns/ballistic/laser_gatling.dm
Comment (RDS88-beeman) [inline code/modules/projectiles/guns/ballistic/laser_gatling.dm]: missing a "To" here, although i to do not want my hands melt

**Comment by KazooBard** [comment]
Looks great! I like the amount of mobs and substantial rewards. Now, if I am correct ti looks as if it were a multi-choice map where YOU choose where to go. Now, about cheesing. I cheesed a LOT of ruins back in the day, and I know a lot about the best ways to do it. It seems there is little to no way to get to the goodies without any obstacles. I see the triple/quadruple walls are the deluxe anti-cheese tool, and I like how you did somethign about it. I like all of the loot if not most is guarded by good NPC's. It looks wacky, fun and still interesting. Good job my dude, I hope cae adds this!

File: code/modules/projectiles/guns/ballistic/laser_gatling.dm
Comment (OceanFish1) [reply code/modules/projectiles/guns/ballistic/laser_gatling.dm]: will get on that later

**Comment by OceanFish1** [comment]
> I hope cae adds this!

Yeah... Sure...

**Comment by KazooBard** [comment]
> > I hope cae adds this!
> 
> Yeah... Sure...

😔

File: code/modules/awaymissions/mission_code/TheFactory.dm
Comment (utilbird) [inline code/modules/awaymissions/mission_code/TheFactory.dm]:
```suggestion
		user.visible_message("<span class='warning'>[user] activates \the [src].</span>", "<span class='notice'>You activate \the [src]!</span>", "<span class='italics'>You hear a music playing.</span>")
```

File: code/modules/awaymissions/mission_code/TheFactory.dm
Comment (utilbird) [inline code/modules/awaymissions/mission_code/TheFactory.dm]: Shouldn't be needed

File: code/modules/awaymissions/mission_code/TheFactory.dm
Comment (utilbird) [inline code/modules/awaymissions/mission_code/TheFactory.dm]:
```suggestion
	visible_message("<span class='boldwarning'>\The [src] activates its self-destruct system!.</span>")
```

File: code/modules/awaymissions/mission_code/TheFactory.dm
Comment (utilbird) [inline code/modules/awaymissions/mission_code/TheFactory.dm]:
```suggestion
		user.visible_message("<span class='rose'>With a single move of [user.p_their()] arm, [user] flips \the [src] and loads the chamber with a shell.</span>")
```

File: code/modules/awaymissions/mission_code/TheFactory.dm
Comment (utilbird) [inline code/modules/awaymissions/mission_code/TheFactory.dm]:
```suggestion
```

File: code/modules/awaymissions/mission_code/TheFactory.dm
Comment (utilbird) [inline code/modules/awaymissions/mission_code/TheFactory.dm]:
```suggestion
```

File: code/modules/awaymissions/mission_code/TheFactory.dm
Comment (utilbird) [inline code/modules/awaymissions/mission_code/TheFactory.dm:795]:
```suggestion
		ranged_cooldown_time = 20//less health - faster shooting
		return
```

File: code/modules/awaymissions/mission_code/TheFactory.dm
Comment (utilbird) [inline code/modules/awaymissions/mission_code/TheFactory.dm]:
```suggestion
```

File: code/modules/awaymissions/mission_code/TheFactory.dm
Comment (utilbird) [inline code/modules/awaymissions/mission_code/TheFactory.dm:779]: You should probably add this on a timer instead of using sleep()

File: code/modules/awaymissions/mission_code/TheFactory.dm
Comment (utilbird) [inline code/modules/awaymissions/mission_code/TheFactory.dm]:
```suggestion
	var/static/list/allowed_projectile_types = list(/obj/item/ammo_casing/shotgun/beanbag,
```

File: code/modules/awaymissions/mission_code/TheFactory.dm
Comment (utilbird) [inline code/modules/awaymissions/mission_code/TheFactory.dm]: Should this be defined in the shooting proc?

File: code/modules/awaymissions/mission_code/TheFactory.dm
Comment (utilbird) [inline code/modules/awaymissions/mission_code/TheFactory.dm:741]: higher speed is slower, just an fyi

File: code/modules/awaymissions/mission_code/TheFactory.dm
Comment (utilbird) [inline code/modules/awaymissions/mission_code/TheFactory.dm]:
```suggestion
	explosion(src,0 ,2 ,3 ,flame_range = 3)
```

File: code/modules/awaymissions/mission_code/TheFactory.dm
Comment (utilbird) [inline code/modules/awaymissions/mission_code/TheFactory.dm]:
```suggestion
```

File: code/modules/awaymissions/mission_code/TheFactory.dm
Comment (utilbird) [reply code/modules/awaymissions/mission_code/TheFactory.dm]:
```suggestion
```

File: code/modules/awaymissions/mission_code/TheFactory.dm
Comment (utilbird) [inline code/modules/awaymissions/mission_code/TheFactory.dm]:
```suggestion
```

File: code/modules/awaymissions/mission_code/TheFactory.dm
Comment (utilbird) [inline code/modules/awaymissions/mission_code/TheFactory.dm]:
```suggestion
	if(Aggro() || stat)
		return
	if(prob(20))
		var/chosen_sound = pick(idle_sounds)
		playsound(src, chosen_sound, 50, FALSE)
```

File: code/modules/awaymissions/mission_code/TheFactory.dm
Comment (utilbird) [inline code/modules/awaymissions/mission_code/TheFactory.dm]:
```suggestion
```

File: code/modules/awaymissions/mission_code/TheFactory.dm
Comment (utilbird) [inline code/modules/awaymissions/mission_code/TheFactory.dm]:
```suggestion
	if(Aggro() || stat)
		return
	if(prob(20))
		var/chosen_sound = pick(idle_sounds)
		playsound(src, chosen_sound, 50, TRUE)
```

File: code/modules/awaymissions/mission_code/TheFactory.dm
Comment (utilbird) [inline code/modules/awaymissions/mission_code/TheFactory.dm]:
```suggestion
```

File: code/modules/awaymissions/mission_code/TheFactory.dm
Comment (utilbird) [inline code/modules/awaymissions/mission_code/TheFactory.dm]:
```suggestion
```

File: code/modules/awaymissions/mission_code/TheFactory.dm
Comment (utilbird) [inline code/modules/awaymissions/mission_code/TheFactory.dm]:
```suggestion
	if(Aggro() || stat)
		return
	if(prob(20))
		var/chosen_sound = pick(idle_sounds)
		playsound(src, chosen_sound, 50, TRUE)
```

File: code/modules/awaymissions/mission_code/TheFactory.dm
Comment (utilbird) [inline code/modules/awaymissions/mission_code/TheFactory.dm]:
```suggestion
```

**Review by utilbird — CHANGES_REQUESTED** [review]
There's a lot of repeatition in mob life() procs, might be worth making a subtype for them

A fair bit of unneeded return statements too xd

File: code/modules/awaymissions/mission_code/TheFactory.dm
Comment (OceanFish1) [reply code/modules/awaymissions/mission_code/TheFactory.dm:741]: I know.

File: code/modules/awaymissions/mission_code/TheFactory.dm
Comment (OceanFish1) [reply code/modules/awaymissions/mission_code/TheFactory.dm]: I mean I was supposed to make each shot a random shell.

Comment (utilbird) [reply code/modules/awaymissions/mission_code/TheFactory.dm]: Yeah, but the actual list can be defined elsewhere. This works though

**Comment by utilbird** [comment]
Should be good to go once the review above is addressed! :)

File: code/modules/awaymissions/mission_code/TheFactory.dm
Comment (OceanFish1) [reply code/modules/awaymissions/mission_code/TheFactory.dm:795]: Didn't I change that already?

File: code/modules/awaymissions/mission_code/TheFactory.dm
Comment (OceanFish1) [reply code/modules/awaymissions/mission_code/TheFactory.dm:779]: I know, but when I try to make a timer for playsound it never works. And I don't think it's a good idea to make a whole new proc for one sound.

File: code/modules/awaymissions/mission_code/TheFactory.dm
Comment (utilbird) [reply code/modules/awaymissions/mission_code/TheFactory.dm:795]: Apparently not

**Comment by OceanFish1** [comment]
well my bruv mate I did add that "return" because when I commited your suggestion it happened so there are now two returns in a row

**Review by utilbird — APPROVED** [review]
Not sure why that happened

**Comment by OceanFish1** [comment]
Summon the code owner...............................

**Comment by OceanFish1** [comment]
@Crossedfall @ike709 So you're owning the code? [REDACTED]

**Review by Crossedfall — APPROVED** [review]
Config changes applied.

**Comment by RDS88-beeman** [comment]
hey elon why'd you revert it

**Comment by utilbird** [comment]
I didn't revert it, I started a revert PR which I didn't end up PRing.

I was going to test merge a revert because of the relatively large amount of complaints I heard about some of the map's design choices. But I didn't really have enough information as to what needed to be changed.

**Comment by Ravellon** [comment]
The size of it makes BYOND shart itself and it tends to fail to fully load.

**Pull request merged**
